cpp/cmake/thirdparty/CUDF_GetArrow.cmake

```diff
@@ -0,0 +1,123 @@
+        GIT_REPOSITORY  https://github.com/apache/arrow.git
+        GIT_TAG         apache-arrow-${VERSION}
+        GIT_SHALLOW     TRUE
+        SOURCE_SUBDIR   cpp
+        OPTIONS         "CMAKE_VERBOSE_MAKEFILE ON"
+                        "ARROW_IPC ON"
+                        "ARROW_CUDA ON"
+                        "ARROW_DATASET ON"
+                        "ARROW_WITH_BACKTRACE ON"
+                        "ARROW_CXXFLAGS -w"
+                        "ARROW_JEMALLOC OFF"
+                        "ARROW_ARMV8_ARCH ${ARROW_ARMV8_ARCH}"
+                        "ARROW_SIMD_LEVEL ${ARROW_SIMD_LEVEL}"
+                        "ARROW_BUILD_STATIC ${ARROW_BUILD_STATIC}"
+                        "ARROW_BUILD_SHARED ${ARROW_BUILD_SHARED}"
+                        "ARROW_DEPENDENCY_USE_SHARED ${ARROW_BUILD_SHARED}"
+                        "ARROW_BOOST_USE_SHARED ${ARROW_BUILD_SHARED}"
+                        "ARROW_BROTLI_USE_SHARED ${ARROW_BUILD_SHARED}"
+                        "ARROW_GFLAGS_USE_SHARED ${ARROW_BUILD_SHARED}"
+                        "ARROW_GRPC_USE_SHARED ${ARROW_BUILD_SHARED}"
+                        "ARROW_PROTOBUF_USE_SHARED ${ARROW_BUILD_SHARED}"
+                        "ARROW_ZSTD_USE_SHARED ${ARROW_BUILD_SHARED}")
+
+
+    set(ARROW_FOUND TRUE)
+    set(ARROW_LIBRARIES "")
+
+    # Arrow_ADDED: set if CPM downloaded Arrow from Github
+    # Arrow_DIR:   set if CPM found Arrow on the system/conda/etc.
+    if(Arrow_ADDED OR Arrow_DIR)
+        if(BUILD_STATIC)
+            list(APPEND ARROW_LIBRARIES arrow_static)
+            list(APPEND ARROW_LIBRARIES arrow_cuda_static)
+        else()
+            list(APPEND ARROW_LIBRARIES arrow_shared)
+            list(APPEND ARROW_LIBRARIES arrow_cuda_shared)
+        endif()
+
+        if(Arrow_DIR)
+            # Set this to enable `find_package(ArrowCUDA)`
+            set(ArrowCUDA_DIR "${Arrow_DIR}")
+            find_package(Arrow REQUIRED QUIET)
+            find_package(ArrowCUDA REQUIRED QUIET)
+        elseif(Arrow_ADDED)
+            # Copy these files so we can avoid adding paths in
+            # Arrow_BINARY_DIR to target_include_directories.
+            # That defeats ccache.
+            file(INSTALL "${Arrow_BINARY_DIR}/src/arrow/util/config.h"
+                 DESTINATION "${Arrow_SOURCE_DIR}/cpp/src/arrow/util")
+            file(INSTALL "${Arrow_BINARY_DIR}/src/arrow/gpu/cuda_version.h"
+                 DESTINATION "${Arrow_SOURCE_DIR}/cpp/src/arrow/gpu")
+            ###
+            # This shouldn't be necessary!
+            #
+            # Arrow populates INTERFACE_INCLUDE_DIRECTORIES for the `arrow_static`
+            # and `arrow_shared` targets in FindArrow and FindArrowCUDA respectively,
+            # so for static source-builds, we have to do it after-the-fact.
+            # 
+            # This only works because we know exactly which components we're using.
+            # Don't forget to update this list if we add more!
+            ###
```

Comment:
Is there a JIRA issue opened that we could link to in order to track this?

Reply:
There is not.

Reply:
Should we open one?